metasfresh-webui-api/src/main/java/de/metas/ui/web/handlingunits/HUDocumentView.java

```diff
@@ -186,16 +243,52 @@ public int getM_Product_ID()
 		return productLV == null ? -1 : productLV.getKeyAsInt();
 	}
 	
+	public String getM_Product_DisplayName()
+	{
+		final JSONLookupValue productLV = (JSONLookupValue)delegate.getFieldValueAsJson(I_WEBUI_HU_View.COLUMNNAME_M_Product_ID);
+		return productLV == null ? null : productLV.getName();
+	}
+
+	public I_M_Product getM_Product()
+	{
+		final int productId = getM_Product_ID();
+		if (productId <= 0)
+		{
+			return null;
+		}
+		return InterfaceWrapperHelper.create(Env.getCtx(), productId, I_M_Product.class, ITrx.TRXNAME_None);
+	}
+	
+	public String getPackingInfo()
+	{
+		final Object packingInfo = delegate.getFieldValueAsJson(I_WEBUI_HU_View.COLUMNNAME_PackingInfo);
+		return packingInfo == null ? null : packingInfo.toString();
+	}
+
 	public int getC_UOM_ID()
 	{
 		final JSONLookupValue uomLV = (JSONLookupValue)delegate.getFieldValueAsJson(I_WEBUI_HU_View.COLUMNNAME_C_UOM_ID);
 		return uomLV == null ? -1 : uomLV.getKeyAsInt();
 	}
-	
+
+	public I_C_UOM getC_UOM()
+	{
+		final int uomId = getC_UOM_ID();
+		if (uomId <= 0)
+		{
+			return null;
+		}
+		return InterfaceWrapperHelper.create(Env.getCtx(), uomId, I_C_UOM.class, ITrx.TRXNAME_None);
+	}
+
 	public BigDecimal getQtyCU()
```

Comment:
when is there a cuQty? always, or just if `isCU()` returns true? what happens if there is no cu qty available?